pkg/lib/models/models.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2019 Cortex Labs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package models
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/cortexlabs/cortex/pkg/lib/sets/strset"
+
+	"github.com/cortexlabs/cortex/pkg/lib/aws"
+)
+
+// IsValidS3Directory checks that the path contains a valid S3 directory for Tensorflow models
+// Must contain the following structure:
+// - 1523423423/ (version prefix, usually a timestamp)
+// 		- saved_model.pb
+//		- variables/
+//			- variables.index
+//			- variables.data-00000-of-00001 (there are a variable number of these files)
+func IsValidS3Directory(path string) bool {
+	listOut, err := aws.ListObjectsExternal(path)
```

Comment:
Just to be safe, I think we should do `path = s.EnsureSuffix(path, "/")` before calling `aws.ListObjectsExternal()` since there could be an edge case where file names collide. E.g. if you have `path/to/1`, `path/to/2`, and `path/today`, I think `ListObjectsExternal("path/to")` will return all three files (but not if we add the trailing `/`)

Reply:
On second thought, I think it would be simpler if we checked `IsS3FileExternal()` for each of the desired files, rather than listing. To save re-creating the s3 client each time, maybe `IsS3FileExternal()` can take in `keys ...string`). `ListObjectsExternal()` could then be deleted.

Reply:
need to check for `containsVariableDataFile`